src/main/kotlin/io/sip3/twig/ce/service/HostService.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018-2020 SIP3.IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.twig.ce.service
+
+import io.sip3.twig.ce.domain.Host
+import io.sip3.twig.ce.repository.HostRepository
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.stereotype.Component
+
+@Component
+open class HostService(private val hostRepository: HostRepository) {
+
+    open fun getAll(): Set<Host> {
+        return hostRepository.findAll().toSet()
+    }
+
+    open fun getByName(name: String): Host {
+        return hostRepository.getByNameIgnoreCase(name)
+    }
+
+    open fun add(host: Host): Host {
+        hostRepository.findByNameIgnoreCase(host.name)?.let {
+            throw DuplicateKeyException("Host with name \"${host.name}\" already exists")
+        }
+
+        return hostRepository.save(host)
+    }
+
+    open fun update(host: Host): Host {
+        return hostRepository.getByNameIgnoreCase(host.name).let { existent ->
+            existent.sip = host.sip
+            existent.media = host.media
+            hostRepository.save(existent)
+        }
+    }
+
+    open fun deleteByName(name: String) {
+        hostRepository.getByNameIgnoreCase(name).let { hostRepository.delete(it) }
```

Comment:
If I got it right - this method may throw an exception if host with such name doesn't exist? I think it's totally fine to return 200 every time here

Reply:
Exception will be mapped to `404 Not found`. Actually if no host will be deleted - API can't answer  200 Ok, because there is no host with such name, and nothing was deleted. Answer with 404 will be more informative.

Reply:
Let's assume that this method works as `deleteIfExists`